src/neo/IO/Caching/DataCache.cs

```diff
@@ -80,12 +80,15 @@ public void Commit()
                 {
                     case TrackState.Added:
                         AddInternal(trackable.Key, trackable.Item);
+                        trackable.State = TrackState.None;
                         break;
                     case TrackState.Changed:
                         UpdateInternal(trackable.Key, trackable.Item);
+                        trackable.State = TrackState.None;
                         break;
                     case TrackState.Deleted:
                         DeleteInternal(trackable.Key);
+                        dictionary.Remove(trackable.Key);
```

Comment:
You cannot delete items while traversing.

Reply:
Have modified correspondingly, please have a look